### PR TITLE
Add Capistrano support

### LINKED ISSUE #761: Use Capistrano for deployment
### Descriptive summary

We want to be able to deploy Scholar to scholar-dev.uc.edu, scholar-qa.uc.edu, and scholar.uc.edu without having to use Bamboo.  The best alternative is probably https://capistranorb.com

After we get it working, we can decide if we want to use Capistrano as a backup when Bamboo is down, or just replace Bamboo entirely.

1. Learn about Capistrano
1. Configure it to deploy Scholar to curly.libraries.uc.edu
1. Get it working with the rest of our server environments (do test deploys on scholar-dev and scholar-qa)
1. Create detailed documentation on the Scholar wiki
1. Train the rest of the team how to use it
1. Bonus: investigate the other ways Capistrano can be used for non-deploy tasks (e.g. restarting sidekiq, etc.)

Notes:
* Be careful not to commit any sensitive passwords, keys, etc.
* Ask Glen for access to existing Bamboo deploy scripts

### PR DESCRIPTION
**Pull request by bsp3ars**
Fixes #761

This commit allows us to use the Capistrano tool to do all of our deploys. The official documentation is in the ucrate wiki (https://github.com/uclibs/ucrate/wiki/Deploy-using-Capistrano). The only change outside of deployment that this impacts is the clamav gem. Instead of the commented out line, it was put into a group called production, so when you do future bundles, you'll need to run `bundle install --without production`.

### CODE REVIEW

**Review by hortongn — CHANGES_REQUESTED** [review]
@bsp3ars Go ahead and make the tweaks I suggested in the review and then I will deploy and test again to see if all is well.  Thanks

File: script/start_qa.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+rm -f "$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )" )/log/production.log"
+ln -s /mnt/common/scholar-logs/libschqwl1_production.log "$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )" )/log/production.log"
+touch /mnt/common/scholar-logs/libschqwl1_sidekiq.log
+rm -f "$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )" )/log/sidekiq.log"
+ln -s /mnt/common/scholar-logs/libschqwl1_sidekiq.log "$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )" )/log/sidekiq.log"
+
+ln -s /srv/apps/scholar_capistrano/current /srv/apps/curate_uc 2> /dev/null
+ln -s /srv/apps/scholar-avatars "$(dirname "$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )" )" )/shared/public/systems/avatars" 2> /dev/null
+ln -s /srv/apps/scholar-public-uploads "$(dirname "$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )" )" )/shared/public/uploads/" 2> /dev/null
+ln -s /mnt/common/branding "$(dirname "$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )" )" )/shared/public/branding" 2> /dev/null
```
[the remark this answers is not in the document]
Comment (bsp3ars) [reply script/start_qa.sh]: This should be okay, but we need to check to make sure @hortongn

**Comment by bsp3ars** [comment]
@hortongn The changes have been made, and this is ready for another review/test

**Review by hortongn — CHANGES_REQUESTED** [review]
@bsp3ars There's still a couple of issues with symlinks on QA.

There's no symlink of public/system/avatars to /mnt/common/avatars

There's no symlink of public/uploads to /mnt/common/scholar-public-uploads

It DID properly create the branding symlink, but not uploads or avatars

**Review by hortongn — CHANGES_REQUESTED** [review]
@bsp3ars That fixed the symlink issues with QA.  However, I tried a deploy to scholar-dev and got this failure:

```
03 cp /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/20190920203243/.env.development.local 2> /dev/null
#<Thread:0x00007faaff3cb980@/Users/hortongn/.rvm/gems/ruby-2.5.3@ucrate/gems/sshkit-1.19.1/lib/sshkit/runners/parallel.rb:10 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        1: from /Users/hortongn/.rvm/gems/ruby-2.5.3@ucrate/gems/sshkit-1.19.1/lib/sshkit/runners/parallel.rb:11:in `block (2 levels) in execute'
/Users/hortongn/.rvm/gems/ruby-2.5.3@ucrate/gems/sshkit-1.19.1/lib/sshkit/runners/parallel.rb:15:in `rescue in block (2 levels) in execute': Exception while executing as scholar@localhost: c
p /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/20190920203243/.env.development.local 2> /dev/null exit status: 1 (SSHKit::Runner::ExecuteEr
ror)
cp /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/20190920203243/.env.development.local 2> /dev/null stdout: Nothing written
cp /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/20190920203243/.env.development.local 2> /dev/null stderr: Nothing written
(Backtrace restricted to imported tasks)
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as scholar@localhost: cp /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/2019092020324
3/.env.development.local 2> /dev/null exit status: 1
cp /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/20190920203243/.env.development.local 2> /dev/null stdout: Nothing written
cp /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/20190920203243/.env.development.local 2> /dev/null stderr: Nothing written


Caused by:
SSHKit::Command::Failed: cp /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/20190920203243/.env.development.local 2> /dev/null exit status: 1
cp /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/20190920203243/.env.development.local 2> /dev/null stdout: Nothing written
cp /srv/apps/scholar_capistrano/static/scholar-dev.variables /srv/apps/scholar_capistrano/releases/20190920203243/.env.development.local 2> /dev/null stderr: Nothing written
```

**Comment by bsp3ars** [comment]
@hortongn I wasn't able to duplicate this specific issue, however I did need to change the branch in /config/deploy/dev.rb
After I did this, the deploy was fast and I did not have any issues.

**Comment by hortongn** [comment]
Rats, I forgot to change the branch.  I changed it for qa, but not dev.  Sorry.  I'll try again.

**Review by hortongn — APPROVED** [review]
@bsp3ars Nice work.  I think we need to tweak some of the wiki documentation, but this is ready to merge.

**Pull request merged**